### PR TITLE
Ray pick whitelisting

### PR DESCRIPTION
**Pull request by ericrius1**
This PR adds support for whitelisting entities to be raypicked. If the user provides an array of one or more entityIDs to as the 3rd argument of Entities.findRayIntersection() call, only entities contained in that array will be whitelisted and picked against. Otherwise they will be ignored.

This helps with raypicking performance when user knows there are certain items they want to pick against, and other entities in the scene can be ignored. For example, create a whiteboard in a complex scene such as toybox by runnning whiteboardSpawner.js. Now, in whiteboardEntityScript.js, uncomment line 57 and comment in line 59. The script will now pick against all entities in the scene, including complex models with many thousands of unsorted triangles, twice each frame (most likely at 60 fps- yikes!) as opposed to only the whiteboard. A noticeable increase in judder in the Rift becomes apparent.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: f93b1d3325bd12a1f0f1c8e7a08161f6936dc908 available at: https://gist.github.com/8b926278d1a108165beb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.dmg

File: examples/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -568,7 +569,7 @@ function MyController(hand, triggerAction) {
             this.setState(STATE_RELEASE);
             return;
         }
-
+            print("AHHHHHH")
```
Comment (Atlante45) [inline examples/controllers/handControllerGrab.js]: Interesting debug message choice.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fb040426eb956c4781c730f1dc9ed67c855780e3 available at: https://gist.github.com/8b926278d1a108165beb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 33ff21f3766386cf94f5bbc7be642e25d854c88b available at: https://gist.github.com/8b926278d1a108165beb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 04f0d792bb2460845d1cf4b917da0edc7295dad5 available at: https://gist.github.com/8b926278d1a108165beb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2f620e654bf6c5a2b8c2466d39f6039fec4e3731 available at: https://gist.github.com/8b926278d1a108165beb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47c3df7587618e5d28f2cd45d33d00d3f88a5fe5 available at: https://gist.github.com/8b926278d1a108165beb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6066.dmg

**Pull request closed without merging**